kin-devplatform-sdk/src/main/java/kin/devplatform/network/api/RestorableWalletApi.java

```diff
@@ -0,0 +1,54 @@
+package kin.devplatform.network.api;
+
+import static kin.devplatform.core.network.ApiClient.GET;
+
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import kin.devplatform.ConfigurationImpl;
+import kin.devplatform.core.network.ApiCallback;
+import kin.devplatform.core.network.ApiClient;
+import kin.devplatform.core.network.ApiException;
+import kin.devplatform.network.model.RestorableWalletRequest;
+import okhttp3.Call;
+
+public class RestorableWalletApi {
+
+	private final ApiClient apiClient;
+
+	public RestorableWalletApi() {
+		this(ConfigurationImpl.getInstance().getDefaultApiClient());
+	}
+
+	public RestorableWalletApi(ApiClient apiClient) {
+		this.apiClient = apiClient;
+	}
+
+	public void getIsRestorableWallet(String publicAddress, final ApiCallback<RestorableWalletRequest> apiCallback)
+		throws ApiException {
+		validateKinVersionApiParams(publicAddress);
+		Call call = getRestorableApiCall(publicAddress);
+		Type localVarReturnType = new TypeToken<RestorableWalletRequest>() {
+		}.getType();
+		apiClient.executeAsync(call, localVarReturnType, apiCallback);
+	}
+
+	private void validateKinVersionApiParams(String publicAddress) throws ApiException {
```

Comment:
an exception of this kind should be `IllegalArgumentException` which is runtime error.

Reply:
I copied this kind of functionallity from all other API calss who validate thier params in this way.
For example, 'updateUserValidateBeforeCall', ' signInValidateBeforeCall' and more methods.

Reply:
ok, let's keep it like it is